### PR TITLE
Seq2seq

### PR DESCRIPTION
**Pull request by ArnoldLIULJ**
ADD Stacked Layer Seq2Seq

### CODE REVIEW

File: tensorlayer/models/seq2seq.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+class Seq2seq(Model):
+
+    """vanilla stacked layer Seq2Seq model.
+
+    Parameters
+    ----------
+    decoder_seq_length: int
+        The length of your target sequence
+    cell_enc : str, tf.function
+        The RNN function cell for your encoder stack, i.e. tf.keras.layers.GRUCell
```
Comment (JingqingZ) [inline tensorlayer/models/seq2seq.py]: 'i.e.' means 'in other words'. Please use 'e.g.' (means 'for example') here and other places accordingly.

File: tensorlayer/models/seq2seq.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    cell_dec : str, tf.function
+        The RNN function cell for your decoder stack, i.e. tf.keras.layers.GRUCell
+    n_layer : int
+        The number of your RNN layers for both encoder and decoder block
+    embbedding_layer : tl.function
+        The embedding layer function, i.e. tl.layers.Embedding(vocabulary_size=voc_size, embedding_size=emb_dim)
```
Comment (JingqingZ) [inline tensorlayer/models/seq2seq.py]: Typo here and please change this comment to:
```
embedding_layer : tl.Layer
        A embedding layer, e.g. tl.layers.Embedding(vocabulary_size=voc_size, embedding_size=emb_dim)
```

File: tensorlayer/models/seq2seq.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    is_train : bool
```
Comment (JingqingZ) [inline tensorlayer/models/seq2seq.py]: `is_train` is never used in `init` and it is not going to be supported when a model is constructed in future versions. Please remove it.

File: tensorlayer/models/seq2seq.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    def __init__(
+            self,
+            decoder_seq_length,
+            cell_enc,
+            cell_dec,
+            n_units=256,
+            n_layer=3,
+            embedding_layer=None,
+            is_train=True,
+            name="seq2seq_"
```
Comment (JingqingZ) [inline tensorlayer/models/seq2seq.py]: `name=None` by default. There is an auto-naming mechanism in `tl.models.Model()`.

File: tensorlayer/models/seq2seq.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+            else:
+                self.dec_layers.append(tl.layers.RNN(cell=cell_dec(units=n_units), in_channels=n_units, return_last_state=True))
+
+        self.reshape_layer = tl.layers.Reshape([-1, n_units])
+        self.dense_layer = tl.layers.Dense(n_units=self.vocabulary_size, in_channels=n_units)
+        self.reshape_layer_after = tl.layers.Reshape([-1, decoder_seq_length, self.vocabulary_size])
+        self.reshape_layer_individual_sequence = tl.layers.Reshape([-1, 1, self.vocabulary_size])
+
+    def inference(self, encoding, seq_length, start_token, top_n):
+
```
Comment (JingqingZ) [inline tensorlayer/models/seq2seq.py]: Some documentation is needed here.

Comment (JingqingZ) [reply tensorlayer/models/seq2seq.py]: Did you use beam search? Or just random sampling?

Comment (ArnoldLIULJ) [reply tensorlayer/models/seq2seq.py]: Random sampling

**Pull request merged**